### PR TITLE
Fix #287 - Add component Tab

### LINKED ISSUE #287: New tab for generic objects
Is it possible to add a new Components tab like the Computers Assets

Components

### PR DESCRIPTION
**Pull request by keha35**
Here is a PR for adding the component tab in objects

### CODE REVIEW

File: setup.php [hunk 1]
```diff
@@ -35,7 +35,7 @@
  ----------------------------------------------------------------------
  */
 
-define ('PLUGIN_GENERICOBJECT_VERSION', '2.9.2');
+define ('PLUGIN_GENERICOBJECT_VERSION', '2.9.3');
```
Comment (cedric-anne) [inline setup.php:38]: Please not change version inside a PR. This is done during the release process.
```suggestion
define ('PLUGIN_GENERICOBJECT_VERSION', '2.9.2');
```

File: inc/object.class.php [hunk 2]
```diff
@@ -174,7 +174,8 @@ static function registerType() {
          "reservation_types"      => $item->canBeReserved(),
          "contract_types"         => $item->canUseContracts(),
          "unicity_types"          => $item->canUseUnicity(),
-         "location_types"         => isset($fields['locations_id'])
+         "location_types"         => isset($fields['locations_id']),
+         "itemdevices_types"         => $item->canUseItemDevice()
```
Comment (cedric-anne) [inline inc/object.class.php:178]:
```suggestion
         "itemdevices_types"      => $item->canUseItemDevice()
```

File: inc/type.class.php [hunk 3]
```diff
@@ -541,6 +549,7 @@ function showBehaviorForm($ID, $options = []) {
             "use_global_search" => __("Global search"),
             "use_projects"      => _n("Project", "Projects", 2),
             "use_network_ports" => __("Network connections", "genericobject"),
+            "use_itemdevices" => _n('Component', 'Components', 2),
```
Comment (cedric-anne) [inline inc/type.class.php:552]:
```suggestion
            "use_itemdevices"   => _n('Component', 'Components', 2),
```

**Review by cedric-anne — COMMENTED** [review]
Seems OK.

**Comment by cedric-anne** [comment]
I switched target branch to develop.

**Review by orthagh — APPROVED** [review]

**Pull request merged**